src/ErrorHandler.php

```diff
@@ -363,10 +363,10 @@ public function addExceptionHandlerListener(callable $listener): void
      */
     private function handleError(int $level, string $message, string $file, int $line, ?array $errcontext = []): bool
     {
-        if (0 === error_reporting()) {
-            $errorAsException = new SilencedErrorException(self::ERROR_LEVELS_DESCRIPTION[$level] . ': ' . $message, 0, $level, $file, $line);
-        } else {
+        if ($level & error_reporting()) {
             $errorAsException = new \ErrorException(self::ERROR_LEVELS_DESCRIPTION[$level] . ': ' . $message, 0, $level, $file, $line);
+        } else {
+            $errorAsException = new SilencedErrorException(self::ERROR_LEVELS_DESCRIPTION[$level] . ': ' . $message, 0, $level, $file, $line);
```

Comment:
This is something different. A `SilencedErrorException` is something like `@foo()`, related to the `@` silencing operator.

Reply:
I see that it was designed as related to `@`. My point is that the case of intentionally silenced by tweaking `error_reporting` warnings should be handled the same way.
I see that Sentry has its own reporting level. Let's review how it should work in such particular case:
- I have `error_reporting=E_ALL`.
- My Sentry error reporting is set to `WARNING`: I want to see in my Sentry dashboard everything that is equals or above `WARNING`.
- Symfony decides to workaround missed cache issue by tweaking `error_reporting` level. The idea is to intentionally "skip" particular warning. I don't want to see it in my Sentry dashboard, it was silenced - just by another way than `@`, see https://github.com/symfony/symfony/blob/6db274e908ae1991f71ce6bfdcaa88c213f4dcf2/src/Symfony/Component/HttpKernel/Kernel.php#L787 .
In my view tweaking `error_reporting` level should be considered as another way of intentionally silencing `@`.

> This is in fact a BC, we cannot accept it as a patch.
Ok, I understand. I am going to rebase it to another branch once we reconcile the most appropriate way of dealing with this issue.

Reply:
> Applying this change would mean intersecting the two settings.

You made me think, thank you :)
I think there is no intersection:
- Sentry `Options::getErrorTypes()` controls whether to report or not.
- Globally defined `error_reporting` impacts how to report the error: either silenced or usual way.
- There is another option (`setCaptureSilencedErrors`), which makes to report even silenced errors.

In my view it works almost the same way as we already have: just the case of `0 === error_reporting()` was expanded to `$level & error_reporting()`. Isn't it?:)